vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -229,13 +238,15 @@ def invoke_fused_moe_kernel(A: torch.Tensor, B: torch.Tensor, C: torch.Tensor,
                             num_tokens_post_padded: torch.Tensor,
                             mul_routed_weight: bool, top_k: int,
                             config: Dict[str, Any], compute_type: tl.dtype,
-                            use_fp8: bool) -> None:
+                            use_fp8: bool, use_int8: bool) -> None:
     assert topk_weights.stride(1) == 1
     assert sorted_token_ids.stride(0) == 1
 
-    if not use_fp8:
+    if not use_fp8 and not use_int8:
         assert A_scale is None
         assert B_scale is None
+    elif use_int8:
+        assert B_scale is not None
     else:
```

Comment:
Maybe it is best to reverse this ordering so `if not use_fp8 and not use_int8` just becomes the `else` case

Reply:
Addressed :)